assets/src/stories-editor/components/block-mover/block-draggable.js

```diff
@@ -54,13 +90,108 @@ BlockDraggable.propTypes = {
 	children: PropTypes.any.isRequired,
 	onDragStart: PropTypes.func,
 	onDragEnd: PropTypes.func,
+	onNeighborHover: PropTypes.func,
+	onNeighborDrop: PropTypes.func,
+	getNeighborPageId: PropTypes.func,
 };
 
-export default withSelect( ( select, { clientId } ) => {
-	const { getBlockIndex, getBlockRootClientId } = select( 'core/block-editor' );
-	const rootClientId = getBlockRootClientId( clientId );
-	return {
-		index: getBlockIndex( clientId, rootClientId ),
-		rootClientId,
-	};
-} )( BlockDraggable );
+export default compose(
+	withSelect( ( select, { clientId } ) => {
+		const { getBlockIndex, getBlockRootClientId, getBlockOrder, getBlock } = select( 'core/block-editor' );
+		const rootClientId = getBlockRootClientId( clientId );
+		return {
+			index: getBlockIndex( clientId, rootClientId ),
+			rootClientId,
+			block: getBlock( clientId ),
+			getBlockOrder,
+		};
+	} ),
+	withDispatch( ( dispatch, { getBlockOrder, rootClientId, clientId, block, blockName } ) => {
+		const { setCurrentPage } = dispatch( 'amp/story' );
+		const { selectBlock, removeBlock, insertBlock, updateBlockAttributes } = dispatch( 'core/block-editor' );
+
```

Comment:
Why not use `ALLOWED_MOVABLE_BLOCKS` here and if not a movable block, then don't do the move.

Reply:
Unmovable blocks aren't wrapped in `Draggable`, so this code will never be run for those - AFAIK.

Reply:
Good point. Your check below for CTA confused me, as that is a none draggable block.

Reply:
(Not sure if it's relevant here but FI that the button inside CTA block is draggable and it does use the `StoryBlockMover` component.)

Reply:
Ah, I see. The button inside the CTA block is the draggable one, but for the purposes of the draggable component, it kind of appears to be the block that is draggable. There are some checks around this making sure that it's the right part of the block that moves of course.